### PR TITLE
fix regeneration in bazel_to_go

### LINKED ISSUE #1515: Bazel_to_go should copy genfiles only on change and maintain go dep file
Bazel_to_go already processes the workspace.

1. It copies generated files and updates a manifest.
2. Creates symlinks from vendors dir

It should 
1. Only copy gen files if they have meaningfully changed.
 The generated files have the absolute source file path embedded in them.
 This makes it susceptible to spurious changes. Tool should ignore change if it is purely path change in the genfile.
2. It should authoritatively generate and maintain the go dep file.

@costinm @jmuk

### PR DESCRIPTION
**Pull request by jmuk**
**What this PR does / why we need it**:

bazel_to_go currently scans `bazel-genfiles` directory for the generated files, but this doesn't work for
presubmit bot, since it shares the build cache among runs (i.e. some generated files in a presubmit run
may appear in another run of a totally unrelated PR).

This PR changes this situation:
- use `bazel query` to list up the files which should be generated in the current workspace.
- run `bazel build`, so it can find the source files regardless of the current build status.
- split the copying of the regenerated files into another file -- and pre-commit does the regeneration only.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1515

**Special notes for your reviewer**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: bin/regenerate_files.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    for (src, dest) in generated_files:
+        try:
+            if should_copy(src, dest):
+                shutil.copyfile(src, dest)
+        except Exception as ex:
+            print src, dest, ex
+
+def main(args):
+    WKSPC = os.getcwd()
```
Comment (kyessenov) [inline bin/regenerate_files.py]: this could be retrieved with bazel info

Comment (jmuk) [reply bin/regenerate_files.py]: Done. thanks!

File: bin/regenerate_files.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    return has_diff
+
+def get_generated_files(WKSPC, genfiles):
+    lst = []
+    proto_libraries = 'kind("go_proto_library", "//...")'
+    proto_compiles = 'attr("langs", "go", kind("proto_compile", "//..."))'
+    genrules = 'attr("outs", ".go", kind("genrule", "//..."))'
+    bazel_util.bazel_build(bazel_util.bazel_query('+'.join([proto_libraries, proto_compiles, genrules])))
+
+    for proto in bazel_util.bazel_query('filter("\.proto$", labels("srcs", %s))' % proto_libraries):
```
Comment (kyessenov) [inline bin/regenerate_files.py]: I suggest adding examples of input -> output for each case, in the very likely case, bazel breaks this again.

**Comment by jmuk** [comment]
@kyessenov can you also make the github approval too?

**Pull request merged**